orangecontrib/text/pubmed.py

```diff
@@ -0,0 +1,480 @@
+            'data' and m is the number of metadata fields we've chosen to
+            extract from the xml. The variable class_values is a list, where
+            the elements are class values for each article instance.
+    """
+    class_values = []
+    metadata = np.empty((len(records), len(includes_metadata)), dtype=object)
+
+    for row_num, record in enumerate(records):
+        fields = [
+                _join_if_list(record.get(field_key, ''))
+                for field_name, field_key
+                in includes_metadata
+            ]
+        fields[-1] = _date_to_iso(fields[-1])
+        metadata[row_num] = np.array(fields)[None, :]
+
+        class_values.append(_mesh_headings_to_class(record.get('MH')))
+    return metadata, class_values
+
+
+def _corpus_from_records(records, includes_metadata):
+    """Receives PubMed records and transforms them into a corpus.
+
+    Args:
+        records (list): A list of PubMed entries.
+        includes_metadata (list): A list of text fields to include.
+
+    Returns:
+        corpus: The output Corpus.
+    """
+    meta_values, class_values = _records_to_corpus_entries(
+            records,
+            includes_metadata=includes_metadata
+    )
+    meta_vars = [
+        StringVariable.make(field_name)
+        for field_name, field_key
+        in includes_metadata
+    ]
+    class_vars = [
+        DiscreteVariable(
+                'section_name',
+                values=list(set(class_values))
+        )
+    ]
+    domain = Domain(
+            [],
+            class_vars=class_vars,
+            metas=meta_vars
+    )
+
+    Y = np.array([class_vars[0].to_val(cv)
+                  for cv
+                  in class_values])[:, None]
+    Y[np.isnan(Y)] = 0
+
+    return Corpus(
+            None,
+            Y,
+            meta_values,
+            domain
+    )
+
+
+class Pubmed:
+
+    DEFAULT_BATCH_SIZE = 1000
+
+    def __init__(self, email, progress_callback=None, error_callback=None):
+        if not validate_email(email):
+            raise ValueError('{} is not a valid email address.'.format(email))
+
+        self.email = email
+        Entrez.email = email
+
+        self.record_id_list = None  # Ids of the records available.
+        self.search_record_count = 0  # Number of records.
+        self.search_record_web_env = None  # BioPython history param.
+        self.search_record_query_key = None  # BioPython history param.
+
+        self.progress_callback = progress_callback
+        self.error_callback = error_callback
+
+        self.cache_path = None
+        cache_folder = os.path.join(environ.buffer_dir, 'pubmedcache')
+        try:
+            if not os.path.exists(cache_folder):
+                os.makedirs(cache_folder)
+            self.cache_path = os.path.join(cache_folder, 'query_cache')
+        except:
+            warnings.warn('Could not assemble Pubmed query cache path',
```

Comment:
Is the problem really in joining paths, or is it rather in creating a folder. What about `Could not initialise Pubmed cache` ?